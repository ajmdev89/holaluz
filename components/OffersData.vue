<template>
  <div class="text-black">
    <h2 class="w-full text-4xl text-center mb-8">¡Enhorabuena!</h2>
    <p v-if="offer.discount > 0" class="w-full text-4xl text-center">Podemos ofercerte un descuento del <span class="font-bold text-green-700">{{ offer.discount }}%</span></p>
    <p v-else class="w-full text-4xl text-center">Cumples las condiciones para unirte a la revolución de los tejados</p>

    <h2 class="text-2xl mt-8">Datos del suministro:</h2>
    <ul class="my-4 flex flex-col md:flex-row">
      <li class="w-full md:w-1/2">CUPS: <span class="font-bold">{{ offer.cups }}</span></li>
      <li class="w-full md:w-1/2">Dirección: <span class="font-bold">{{ offer.address }}</span></li>
    </ul>
  </div>
</template>

<script>
export default {
  name: 'OffersData',

  props: {
    offer: {
      type: Object,
      required: true
    }
  }
}
</script>