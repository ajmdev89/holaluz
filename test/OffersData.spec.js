import { shallowMount } from '@vue/test-utils'
import OffersData from '~/components/OffersData.vue'

const mockData = {
  discount: 1,
  cups: 1234,
  address: 'foo'
}

const mockDataNoDiscount = {
  discount: 0,
  cups: 1234,
  address: 'foo'
}

describe('OffersData component -----OffersData.vue-----', () => {
  let wrapper

  it('is a Vue instance', () => {
    const wrapper = shallowMount(OffersData, { propsData: { offer: mockData } })
    expect(wrapper.vm).toBeTruthy()
  })

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('it should render offer data with no discount', () => {
    const wrapper = shallowMount(OffersData, { propsData: { offer: mockDataNoDiscount } })
    const messageWrapper = wrapper.find('p')

    expect(messageWrapper.text()).toBe('Cumples las condiciones para unirte a la revoluciÃ³n de los tejados')
  })

  it('it should render offer data with discount', () => {
    const wrapper = shallowMount(OffersData, { propsData: { offer: mockData } })
    const messageWrapper = wrapper.find('span')

    expect(messageWrapper.text()).toBe(`${mockData.discount}%`)
  })
})
