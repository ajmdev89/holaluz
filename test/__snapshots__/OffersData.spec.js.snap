// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OffersData component -----OffersData.vue----- matches snapshot 1`] = `undefined`;
